db/mongodb.go

```diff
@@ -0,0 +1,283 @@
+		"value": v,
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	_, err := collection.InsertOne(ctx, doc)
+	if err != nil {
+		return fmt.Errorf("error saving %s to the meta collection: %w", k, err)
+	}
+	return nil
+}
+
+// MetaRead reads a key/value pair from the metadata collection.
+func (m *MongoDB) MetaRead(k string) (string, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	var result struct {
+		Value string `bson:"value"`
+	}
+	collection := m.db.Collection(metaTableName)
+	err := collection.FindOne(ctx, bson.M{"key": k}).Decode(&result)
+	if err != nil {
+		if err == mongo.ErrNoDocuments {
+			return "", fmt.Errorf("metadata key %s not found", k)
+		}
+		return "", fmt.Errorf("error looking for metadata key %s: %w", k, err)
+	}
+	return result.Value, nil
+}
+
+// Close terminates the connection to MongoDB.
+func (m *MongoDB) Close() error {
+	if err := m.client.Disconnect(m.ctx); err != nil {
+		return fmt.Errorf("error disconnecting from MongoDB: %w", err)
+	}
+	fmt.Println("Successfully disconnected from MongoDB")
+	return nil
+}
+
+// PreLoad runs before starting to load data into the database.
+func (m *MongoDB) PreLoad() error {
+	return nil
+}
+
+// PostLoad runs after loading data into the database.
+// Removes duplicates and creates indexes.
+func (m *MongoDB) PostLoad() error {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	collection := m.db.Collection(companyTableName)
+	pipeline := mongo.Pipeline{
+		{{"$group", bson.D{
+			{"_id", "$cnpj"},
+			{"docs", bson.D{{"$push", "$_id"}}},
+			{"count", bson.D{{"$sum", 1}}},
+		}}},
+		{{"$match", bson.D{{"count", bson.D{{"$gt", 1}}}}}},
+	}
+	cursor, err := collection.Aggregate(ctx, pipeline)
+	if err != nil {
+		return fmt.Errorf("error executing aggregation: %w", err)
+	}
+	defer cursor.Close(ctx)
+	// Iterates through the results and removes duplicates
+	for cursor.Next(ctx) {
+		var result struct {
+			ID   string               `bson:"_id"`
+			Docs []primitive.ObjectID `bson:"docs"`
+		}
+		if err := cursor.Decode(&result); err != nil {
+			return fmt.Errorf("error decoding result: %w", err)
+		}
+		// Keep the first document and remove the others
+		if len(result.Docs) > 1 {
+			toRemove := result.Docs[1:] // Delete the first document
+			_, err := collection.DeleteMany(ctx, bson.M{"_id": bson.M{"$in": toRemove}})
+			if err != nil {
+				return fmt.Errorf("error removing duplicates: %w", err)
+			}
+		}
+	}
+	if err := cursor.Err(); err != nil {
+		return fmt.Errorf("error when iterating through results: %w", err)
+	}
+	if err := m.CreateIndexes(); err != nil {
+		return fmt.Errorf("error creating indexes: %w", err)
+	}
+	return nil
+}
+
+func (m *MongoDB) GetCompany(cnpj string) (string, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	collection := m.db.Collection(companyTableName)
+	filter := bson.M{"cnpj": cnpj}
+	var empresa empresa
```

Comment:
Praticamente na base toda usamos `c` de _company_

```suggestion
	var c empresa
```